v0.5/fastn-p1/src/parser/scanner.rs

```diff
@@ -105,11 +105,13 @@ impl Scanner {
 
     #[cfg(test)]
     pub fn remaining(&self) -> String {
-        let mut s = String::new();
-        for c in &self.tokens[self.index..] {
-            s.push(*c);
-        }
-        s
+        self.tokens[self.index..].iter().collect()
+    }
+
+    #[cfg(test)]
+    pub fn s_remaining(&self) -> String {
+        let input: String = self.tokens.iter().collect();
+        input[self.s_index..].to_string()
```

Comment:
Nopes, don't create this. Make existing remaining() compute do the same and assert both methods produces same strings.